src/CMakeLists.txt

```diff
@@ -90,9 +52,57 @@ add_library(dispatch
               shims/perfmon.h
               shims/time.h
               shims/tsd.h
-              shims/yield.h
-              ${dispatch_BLOCK_SOURCES}
-              ${dispatch_SWIFT_SOURCES})
+              shims/yield.h)
+if(DISPATCH_USE_INTERNAL_WORKQUEUE)
+  target_sources(dispatch
+                 PRIVATE
+                   event/workqueue.c
+                   event/workqueue_internal.h)
+endif()
+target_sources(dispatch
+               PRIVATE
+                 block.cpp)
+if(HAVE_OBJC)
+  target_sources(dispatch
+                 PRIVATE
+                   data.m
+                   object.m)
+endif()
+if(CMAKE_SWIFT_COMPILER)
+  set(swift_optimization_flags)
+  if(CMAKE_BUILD_TYPE MATCHES Release)
+    set(swift_optimization_flags -O)
+  endif()
+  add_swift_library(swiftDispatch
+                    MODULE_NAME
+                      Dispatch
+                    MODULE_LINK_NAME
+                      dispatch
+                    MODULE_PATH
+                      ${CMAKE_CURRENT_BINARY_DIR}/swift/Dispatch.swiftmodule
+                    OUTPUT
+                      ${CMAKE_CURRENT_BINARY_DIR}/swiftDispatch.o
+                    SOURCES
+                      swift/Block.swift
+                      swift/Data.swift
+                      swift/Dispatch.swift
+                      swift/IO.swift
+                      swift/Private.swift
+                      swift/Queue.swift
+                      swift/Source.swift
+                      swift/Time.swift
+                      swift/Wrapper.swift
+                    CFLAGS
+                      -fblocks
+                      -fmodule-map-file=${CMAKE_SOURCE_DIR}/dispatch/module.modulemap
```

Comment:
this is a pre-existing bug but this is the linux modulemap, if we ever want this part to work on Darwin this path will need to be conditionalized per platform